src/lib.rs

```diff
@@ -511,8 +497,8 @@ impl ZooKeeper {
         WithWatcher(self)
     }
 
+    #[instrument]
```

Comment:
question: Do we want to use the same event name as currently? If yes, this would need a slight adjustment (also a few other functions below):

```suggestion
    #[instrument(name = "exists")]
```